cmd/zdb/zdb.c

```diff
@@ -4406,7 +4424,7 @@ main(int argc, char **argv)
 	if (error == 0) {
 		if (target_is_spa || dump_opt['R']) {
 			error = spa_open_rewind(target, &spa, FTAG, policy,
-			    NULL);
+			    NULL, ZPOOL_NO_ACT_CHK);
```

Comment:
By the time spa_lookup() works in zdb, the spa has already been built up from the config nvlist in a function a few layers beneath the kernel_init().  Instead, I added a boolean field to the spa which when true indicates the activity check should be skipped.  I do a spa_lookup and then set the field to B_TRUE.